Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -701,9 +688,8 @@ FieldValue FieldValue::FromString(std::string&& value) {
   return FieldValue(std::make_shared<StringValue>(std::move(value)));
 }
 
-FieldValue FieldValue::FromBlob(const uint8_t* source, size_t size) {
-  std::vector<uint8_t> copy(source, source + size);
-  return FieldValue(std::make_shared<BlobValue>(std::move(copy)));
+FieldValue FieldValue::FromBlob(ByteString blob) {
+  return FieldValue(std::make_shared<BlobValue>(std::move(blob)));
```

Comment:
Can this be `std::make_shared<const BlobValue>`? Feel free to ignore if there's too much collateral compilation damage.